### PR TITLE
Add multi-node distributed training script

### LINKED ISSUE #170: Set up multi-node training pipeline functionality
### Details
Training a large model using model parallelism such as #177  is much slower than training a smaller model in a purely data parallel manner due to the increased inter-device communications incurred by model parallelism.

Multi-node training would mitigate these additional costs and increase pipeline efficiency by allowing us to train across a larger number of machines.

We should explore the maximum size possible to train using the Hugging Face `accelerate` and `torch.distributed` packages on a single Node while also considering swapping back to GPT-NeoX for larger model sizes.

This task is concerned with evaluating different tooling options for multi-node training before implementing these in the training pipeline.

### PR DESCRIPTION
**Pull request by jackapbutler**
* closes #170 

This PR adds a script called `sbatch_train_hf_multinode.sh` which allows for multi-node distributed training for higher throughput on the cluster.

It is worth noting that this setup requires the model to fit inside a single node (320GB VRAM) as it only performs data parallelism across the nodes. There are techniques such as DeepSpeed ZeRO which allow sharding intra-node (as in #189) but it's possible we will require multiple nodes in a model parallel fashion to train larger models efficiently.

### CODE REVIEW

File: experiments/scripts/sbatch_train_hf_multinode.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+# Get multinode information
+nodes=( $( scontrol show hostnames $SLURM_JOB_NODELIST ) )
+nodes_array=($nodes)
+head_node=${nodes_array[0]}
+head_node_ip=$(srun --nodes=1 --ntasks=1 -w "$head_node" hostname --ip-address)
+echo Node IP: $head_node_ip
+
+# Run script
+srun python -m torch.distributed.launch --use-env --nnodes 2 --nproc_per_node 8 \
```
Comment (bethanyconnolly) [inline experiments/scripts/sbatch_train_hf_multinode.sh:43]: I couldnt find `--nnodes` on [here](https://slurm.schedmd.com/srun.html). Maybe this should be `--nodes`?

Comment (jackapbutler) [reply experiments/scripts/sbatch_train_hf_multinode.sh:43]: Ye, for that syntax the argument of `--nnodes` is actually related to the `torch.distributed.launch` launcher script and not `srun`. The ordering is roughly;
```bash
srun <srun-args> <my-launcher> <my-launcher-args> <my-script> <my-script-args>
```

Then essentially all the `srun` parameters are already set up at the top such as 
```bash
#SBATCH --nodes=2
```

It's supposedly best practice to run these larger multinode scripts using the `srun` prefix as this allows the slurm scheduler to track it more effectively and give you more accurate analytics for the job.

**Review by bethanyconnolly — APPROVED** [review]
